CurrencyAlert/Configuration.cs

```diff
@@ -1,38 +1,57 @@
-﻿using CurrencyAlert.Helper;
-using CurrencyAlert.Provider;
-using Dalamud.Configuration;
+﻿using Dalamud.Configuration;
 using Dalamud.Plugin;
 using System;
-using System.Collections.Generic;
+using CurrencyAlert.DataModels;
+using CurrencyAlert.Windows;
+using KamiLib.Configuration;
 
-namespace CurrencyAlert
-{
-    [Serializable]
-    public class Configuration : IPluginConfiguration
-    {
-        public int Version { get; set; } = 5;
-
-        public bool UiLocked { get; set; } = false;
-        public Dictionary<int, bool> AlertEnabled { get; } = new Dictionary<int, bool>();
-        public Dictionary<int, int> Threshold { get; } = new Dictionary<int, int>();
+namespace CurrencyAlert;
 
-        public Configuration()
-        {
-            foreach (var currency in CurrencyProvider.Instance.GetAll())
-            {
-                this.AlertEnabled[currency.Id] = true;
-                this.Threshold[currency.Id] = currency.DefaultThreshold;
-            }
-        }
+[Serializable]
+public class Configuration : IPluginConfiguration
+{
+    public int Version { get; set; } = 6;
 
-        public void Initialize()
-        {
-           
-        }
+    public OverlaySettings OverlaySettings = new();

```

Comment:
I guess it does mean a configuration wipe for users ?

Reply:
Yes, moving forwards there shouldn't be any wipes as you can simply check the array for the enum and add it if it's not there. 

It would be possible to migrate but there are new settings that I think users should be encouraged to look at. For example there's the new chat notification that reminds you via chat to spend the currencies.

Reply:
The plugin uses reasonable defaults, so if someone doesn't want to bother they don't have to.
If you want to draw peoples attention to it, we could make Chat Notifications on by default